src/Concordium/Client/Output.hs

```diff
@@ -1096,7 +1101,37 @@ showRejectReason verbose = \case
     Types.NonExistentTokenId tokenId -> printf "token id %s does not exist on-chain" (show tokenId)
     Types.TokenHolderTransactionFailed reason -> do
         let details = Types.tmrrDetails reason
-            invalidCBOR = printf "token holder transaction failed: %s (Could not decode reject details as valid CBOR)\n" (show reason)
+            invalidCBOR =
+                printf
+                    "%s token-holder transaction was rejected due to: %s; details (undecoded CBOR): %s"
+                    (show $ Types.tmrrTokenSymbol reason)
+                    (show $ Types.tmrrType reason)
+                    (show details)
+        case details of
+            Nothing -> invalidCBOR
+            Just detail -> do
```

Comment:
I think it would be better to get rid of the `Maybe` part when showing the details, e.g.:
```suggestion
        case details of
            Nothing -> printf "%s token-holder transaction was rejected due to: %s" 
                        (show $ Types.tmrrTokenSymbol reason)
                        (show $ Types.tmrrType reason)
            Just detail -> do
                let invalidCBOR =
                    printf
                        "%s token-holder transaction was rejected due to: %s; details (undecoded CBOR): %s"
                        (show $ Types.tmrrTokenSymbol reason)
                        (show $ Types.tmrrType reason)
                        (show detail)
```